### ISSUE TITLE
Trying IPv6 sockets w/ IPv6 being available but down, leads to connection failures

### ISSUE THREAD

**Issue by erikarn** [rofl0r/microsocks#72]
This is ... kinda silly? I'm going to work around it for now by using -b <ip> but it would be nice to update socket logic to try different hosts.

The TL;DR is:

* resolve() returns the host list
* connect_socks_target() only tries /one/ entry in the list
* github.com returns only A records
* freebsd.org git repo IPs return AAAA and A records
* my interfaces have local v6 addresses but currently (for stupid reasons) no global v6 addresses
* so, if I try connecting to freebsd.org hosts, it will see AAAA is likely fine, bind() ok, connect() hits "HOST_UNREACHABLE" and bam.

It would be nice if instead it could loop over the hosts returned by getaddrinfo() and try them until one succeeds?

**Comment by rofl0r** [comment]
you could also patch resolve to use either AF_INET or AF_INET6 for hints instead of AF_UNSPEC.
the code kinda expects that one has his internet connectivity properly set up. maybe i'll take a look at implementing a loop here, but it's the first time anyone complains.

**Comment by erikarn** [comment]
Yeah, I'm fixing it up right now to try the returned list in a loop. Stay
tuned!

(Also yeah I also dislike when internet isn't "properly set up" but somehow
I'm not getting a v6 lease, only a v4. I need to go figure THAT out too.)
(And yeah, my LAN stuff also has v6 enabled!)



-adrian

**Comment by nerozero** [comment]
I guess this issue is still present.
 - Connecting to the proxy from IPv6 host is not possible to me, server simply does not respond
 - Also impossible to force listening on IPv6 IP ( -i <ipv6> )

**Comment by rofl0r** [comment]
> I guess this issue is still present.

this issue is specifically about a misconfigured ipv6, so yes - it's still open, and no - it's not what you experience

>    Connecting to the proxy from IPv6 host is not possible to me, server simply does not respond
>    Also impossible to force listening on IPv6 IP ( -i )

that's what the -b bind directive is for. try it out.
